### PR TITLE
LPS-175935 Render edit toggle button for AC Segments that has a NOT empty criteria

### PR DESCRIPTION
**Pull request by cgoncas**
Motivation
=======
Edit AC Event Segments from DXP

[Allow AC Segments to be edited.](https://issues.liferay.com/browse/LPS-175935)

Proposed Solution
========
Check if the FF feature.flag.LPS-172194 is set to true, in this case only empty criteria segments won't be editable.

Steps to Verify:
----------------
1. Merge this PR in top of https://github.com/liferay-tango/liferay-portal/pull/3226
2. Set feature.flag.LPS-172194 to true
3. Click view an AC Segment Editor
4. The Edit Toggle should be visible.

Screencaps:
----------------

Edit Toggle is available

Captura de pantalla 2023-02-16 a las 14 40 25

When the Edit toggle is checked then the segment can be edited

Captura de pantalla 2023-02-16 a las 14 40 32

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:tango-acceptance

**Review by claraizquierdo — COMMENTED** [review]
Hi @cgoncas ,

I've been testing and I'm able to edit AC segments with the FF set to true and to false 🤔  I don't know if I missing something.

**Comment by cgoncas** [comment]
> Hi @cgoncas ,
> 
> I've been testing and I'm able to edit AC segments with the FF set to true and to false 🤔 I don't know if I missing something.

I think I broke it in my last SF, sorry :(, can you test it again?

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by Mikellorza** [comment]
Hey @cgoncas ,

That is working for me as you can see in the next screenshots:

image
image

Only yo remind that also we will have to build the link of the list to the view mode, I am referring to this: 

image


Thanks

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/security/permission/resource/SegmentsEntryModelResourcePermissionWrapper.java [hunk 1]
```diff
@@ -76,7 +78,10 @@ public Boolean contains(
 
 			if (actionId.equals(ActionKeys.UPDATE) &&
 				SegmentsEntryConstants.SOURCE_ASAH_FARO_BACKEND.equals(
-					segmentsEntry.getSource())) {
+					segmentsEntry.getSource()) &&
+				(!GetterUtil.getBoolean(
+					PropsUtil.get("feature.flag.LPS-172194")) ||
+				 (segmentsEntry.getCriteria() == null))) {
```
Comment (Mikellorza) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/security/permission/resource/SegmentsEntryModelResourcePermissionWrapper.java:84]: Hey @cgoncas ,

This is returning an empty String when it is null as you can see in the impl :  

	public String getCriteria() {
		if (_criteria == null) {
			return "";
		}
		else {
			return _criteria;
		}
	}

So maybe we have to change it.

Thanks

Comment (cgoncas) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/security/permission/resource/SegmentsEntryModelResourcePermissionWrapper.java:84]: Totally true, changed!!

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by claraizquierdo** [comment]
ci:reevaluate:1377401_9141

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
Hey @CarlosBrichete snd @rafaellajordao !!!

could you take a look to the failing tests?

thanks.

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward-force

**Pull request closed without merging**